### ISSUE TITLE
Broken `pyperformance` benchmarks

### ISSUE THREAD

**Issue by brandtbucher** [faster-cpython/ideas#311]
Broken:
- `sqlalchemy_declarative`
- `sqlalchemy_imperative`

Cause: https://bugs.python.org/issue46836

Fixes:
- [x] https://github.com/cython/cython/pull/4667
- [x] https://github.com/python-greenlet/greenlet/pull/294
- [ ] https://github.com/gevent/gevent/pull/1872

**Comment by vstinner** [comment]
> genshi

What is the issue with genshi?

**Comment by brandtbucher** [comment]
> What is the issue with genshi?

Sorry, that was an unrelated issue that has since been fixed. Removed it from the list!

**Comment by gvanrossum** [comment]
See also https://github.com/faster-cpython/ideas/issues/333 for Windows breakages.

**Comment by gvanrossum** [comment]
Seems a new Genshi issue has popped up, see https://github.com/faster-cpython/ideas/issues/333#issuecomment-1076479303 and ff.

**Comment by gvanrossum** [comment]
Well, it wasn't a new genshi issue -- it's fixed in master but there's no release yet.

**Comment by vstinner** [comment]
My greenlet PR is merged https://github.com/python-greenlet/greenlet/pull/294 but there is no release yet.

Problem: it's unclear to me if gevent uses or can use the newer greenlet. It seems like greenlet got a new "v2" design which better hides its internals and CPython internals.

**Comment by vstinner** [comment]
Currently, on PyPI, there are only greenlet 1.x releases. The master branch of greenlet is not released yet. Maybe we change should be re-done in a greenlet "stable" branch, I didn't check if it exists.

**Comment by brandtbucher** [comment]
> Currently, on PyPI, there are only greenlet 1.x releases. The master branch of greenlet is not released yet. Maybe we change should be re-done in a greenlet "stable" branch, I didn't check if it exists.

[`2.0.0a2`](https://pypi.org/project/greenlet/2.0.0a2/) was released yesterday. I believe it includes your changes?

If so, we can just use that. `pyperformance` already using `2.0.0a1`, currently.

**Comment by brandtbucher** [comment]
The annoying thing is that the benchmarks don't actually use `greenlet` (it's just a required dependency of `SQLAlchemy`). For example, just using an old cached 3.11 wheel works fine. 🤦🏼‍♂️

@vstinner, do you want to bump the version in `pyperformance`, or should I?

**Comment by gvanrossum** [comment]
Maybe involve @ericsnowcurrently too? He recently (this week) did a bunch of work, but I’m not sure he is done. The psutil thing isn’t fixed on Windows, and —python is broken.

**Comment by vstinner** [comment]
I no longer maintain pyperformance.

**Comment by gvanrossum** [comment]
Then who does?

**Comment by vstinner** [comment]
> Then who does?

The latest release was done by @pablogsal: https://github.com/python/pyperformance/commit/034f58be960ee88010b7a8046265ca88f0c35aa4

**Comment by gvanrossum** [comment]
Interestingly, for me (on Windows) moving to `greenlet==2.0.0a2` (from `...a1`) the sqlalchemy benchmarks run fine. So perhaps this is done? (Then what is https://github.com/python-greenlet/greenlet/pull/294 about?)

**Comment by gvanrossum** [comment]
Now I'm confused. Your open checkbox links to https://github.com/gevent/gevent/pull/1872 which is still open. What is that then?

Anyway, I confirmed that on macOS all BMs work. So maybe you can close this, or check that box?

**Comment by brandtbucher** [comment]
> Now I'm confused. Your open checkbox links to [gevent/gevent#1872](https://github.com/gevent/gevent/pull/1872) which is still open. What is that then?

Ah, I think that might be actually breaking the *Pyston* macrobenchmarks, not `pyperformance`. I might have added it here since the cause was the same as the others, and there was an open PR to fix it. I can track it elsewhere.

Closing, since I've confirmed locally (Linux) that the `main` branch of `pyperformance` is now working. We still need a new release, though.
